pop/service/contract.go

```diff
@@ -0,0 +1,215 @@
+package service
+
+import (
+	"crypto/sha256"
+	"errors"
+	"fmt"
+
+	"github.com/dedis/cothority"
+	"github.com/dedis/cothority/omniledger/contracts"
+	"github.com/dedis/cothority/omniledger/darc"
+	"github.com/dedis/cothority/omniledger/darc/expression"
+	ol "github.com/dedis/cothority/omniledger/service"
+	"github.com/dedis/kyber"
+	"github.com/dedis/onet/log"
+	"github.com/dedis/onet/network"
+	"github.com/dedis/protobuf"
+)
+
+// This file holds the contracts for the pop-party. The following contracts
+// are defined here:
+//   - PopParty - holds the Configuration and later the FinalStatement
+//   - PopCoinAccount - represents an account of popcoins
+
+// ContractPopParty represents a pop-party that holds either a configuration
+// or a final statement.
+var ContractPopParty = "popParty"
+
+// ContractPopCoinAccount holds popcoins of an attendee or a service.
+var ContractPopCoinAccount = "popCoinAccount"
+
+// PoPCoinName is the identifier of the popcoins.
+var PoPCoinName ol.InstanceID
+
+func init() {
+	h := sha256.New()
+	h.Write([]byte("popcoin"))
+	PoPCoinName = ol.NewInstanceID(h.Sum(nil))
+}
+
+// ContractPopParty represents a pop-party on OmniLedger. It has the following
+// functionalities:
+//   * Spawn - takes a "FinalStatement" argument with the binary representation
+//     of the final statement to store.
+//   * Invoke - hast the following Command
+//     * "Finalize" - that stores a final statement and doesn't let it be
+//       changed afterwards. It will also create a darc for every attendee
+//       and bind that darc to a coin account, putting an initial 1.000.000
+//       popCoins in it.
+//        This command has the following arguments:
+//       * "FinalStatement" - mandatory, to give the new final statement. It
+//         needs to be correctly finalized by the pop-service.
+//       * "Service" - when given, will create a darc and a coin-account for
+//         the service to use.
+func (s *Service) ContractPopParty(cdb ol.CollectionView, inst ol.Instruction, coins []ol.Coin) (scs []ol.StateChange, cOut []ol.Coin, err error) {
+	cOut = coins
+	var darcID darc.ID
+	var ppi PopPartyInstance
+	if inst.Spawn == nil {
```

Comment:
is there a missing signature verification? perhaps it's time to do the contract interface thing so that it'll at least do something by default.